File: src/main/java/com/heybys/oddments/fooddelivery/persistence/order/OrderRepositoryImpl.java
package com.heybys.oddments.fooddelivery.persistence.order;

import org.springframework.stereotype.Repository;

import com.heybys.oddments.base.jpa.BaseRepository;
import com.heybys.oddments.fooddelivery.domain.order.Order;
import com.heybys.oddments.fooddelivery.domain.order.OrderId;
import com.heybys.oddments.fooddelivery.domain.order.OrderRepository;

@Repository
class OrderRepositoryImpl extends BaseRepository<Order, OrderId, OrderJpaRepository> implements OrderRepository {

    public OrderRepositoryImpl(OrderJpaRepository repository) {
        super(repository);
    }
}

File: src/main/kotlin/com/heybys/oddments/fooddelivery/persistence/order/OrderRepositoryImpl.kt
package com.heybys.oddments.fooddelivery.persistence.order

import com.heybys.oddments.base.jpa.BaseRepository
import com.heybys.oddments.fooddelivery.domain.order.Order
import com.heybys.oddments.fooddelivery.domain.order.OrderId
import com.heybys.oddments.fooddelivery.domain.order.OrderRepository
import org.springframework.stereotype.Repository

@Repository
internal class OrderRepositoryImpl(repository: OrderJpaRepository) :
    BaseRepository<Order, OrderId, OrderJpaRepository>(repository), OrderRepository
